### PR TITLE
Upgrade plugin API and embedding to work with Flutter v2

### PR DESCRIPTION
**Pull request by enveraltin**

### CODE REVIEW

**Comment by enveraltin** [comment]
This branch contains @maxzod 's changes as well.

**Review by cedteg — APPROVED** [review]

**Comment by paulo-coutinho** [comment]
MERGE!!!

**Comment by enveraltin** [comment]
For anyone waiting for a merge here, feel free to use my forked repo in your `pubspec.yaml` like this:

```
    advance_pdf_viewer:
        git:
            url: https://github.com/enveraltin/pdf_viewer.git
```

Cheers.

**Comment by Giber05** [comment]
@enveraltin i'm getting force close when i load file from url.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: advance_pdf_viewer
-description: A flutter plugin for handling PDF files. Works on both Android & iOS
-version: 2.0.0
-homepage: https://github.com/lohanidamodar/pdf_viewer
+description: A flutter plugin for handling PDF files. Works on both Android & iOS, forked from https://github.com/lohanidamodar/pdf_viewer to fix compatibility with Flutter 2.2
```
Comment (lohanidamodar) [inline pubspec.yaml:2]:
```suggestion
description: A flutter plugin for handling PDF files. Works on both Android & iOS
```

**Comment by gtu-myowin** [comment]
> > @enveraltin i'm getting force close when i load file from url.
> 
> Are you testing on a device without GMS, by any chance? A Huawei device, perhaps?

@enveraltin 

I just got the same error. The stack trace is mentioned in this issue #86 

My device was Pixel XL API 30 (android-x86 emulator). It was fine on iOS Simulator. Kindly help me since I can neither upgrade nor downgrade. I am stuck for update release.

**Comment by elmariocarlos** [comment]
Hi @lohanidamodar here is the PR request that us not moving forwrd to fix some of the issues;I'll also tag you in another two open issues.

**Pull request merged**